search-commons/src/main/java/no/unit/nva/search2/common/QueryFilter.java

```diff
@@ -25,7 +25,7 @@ public void set(QueryBuilder... filters) {
     }
 
     public void add(QueryBuilder builder) {
-        this.filters.removeIf(filter -> filter.getName().equals(builder.getName()));
```

Comment:
If filters needs to be unique for each queryName you could make filter a HashMap with queryName as key

Reply:
good point